### DISCUSSION TITLE
List view (continued)

### DISCUSSION THREAD

**Discussion by huperaisan** [crocodilestick/Calibre-Web-Automated#121] [Ideas]
There was an [earlier feature request](https://github.com/janeczku/calibre-web/issues/1207) for a compact list view for books, which was closed because "Books List" was implemented. I would like to suggest a couple of changes for the web UI:

- Currently, the "Books List" view does not actually allow you to open a book from it. Clicking on a title lets you change the title instead. Perhaps a button could be added to open books, or perhaps make an "edit mode" toggle.
- There are two columns which aren't removable from the list view (delete and read status). They wouldn't be a problem except they contain buttons with text labels that make the columns very wide. Could these columns be made removable, or at least move the buttons to the top or bottom of the page where they won't expand the columns?
<img width="898" alt="image" src="https://github.com/user-attachments/assets/925f5011-4903-4bd3-af8a-f578123d78fd">

**Comment by Tharic99** [comment]
In addition, the List View page does not appear to respect the browser's width, forcing a bottom horizontal scrollbar to appear rather than simply making the columns less wide.

**Comment by crocodilestick** [comment]
Yeah one of the community members made this feature. Originally the buttons weren't in the column headings and it looked a lot cleaner but as the member was also trying to get the batch editing features added to stock CW, their new head dev asked him to change it to how it currently is 🙄 I'll see if I can dig up his old PR and revert it

**Comment by crocodilestick** [comment]
Hi guys, I've been working on improving the Book List page for everyone today as the state it was in was annoying me too. The new version is much cleaner, much easier to use on smaller screens and mobile and manages to maintain all of the features of the original, just in a much sleeker package. Let me know what you think from the images below and I'll include the improvements as a small update ASAP.

Old Version / Current Version as of V2.1.1:
![old-book-list](https://github.com/user-attachments/assets/0c0a8ee7-4fd9-4731-8d25-76ad596f88f1)

New Proposed Version / Current Dev Build:
![new-book-list](https://github.com/user-attachments/assets/f35e5537-5216-4c87-ab08-91a8c921e5b2)

@qdatk @Tharic99

**Reply by huperaisan** [reply]
Oh I almost forgot: does the new version open the book when you click on the title?

**Comment by Tharic99** [comment]
Looking good. One thing I noticed in the old version and it looks like even in your new version. If you're showing 1-10 of 61 rows, there's no "next" type option to get to the other 51 rows. Plus the # of rows I think (this is from memory) was 100 max, so anything over 100 books and you can't see unless you start sorting different ways. 

So some type of a Page x of y type thing would be needed as well. I know for me personally, I've got a few thousand books in Calibre and just moved over to CWA yesterday.

Thanks!

**Reply by huperaisan** [reply]
> there's no "next" type option to get to the other 51 rows

It's in the top right!

@crocodilestick The new version looks great!

**Comment by viniciuscmartins** [comment]
If I may suggest, what I`m missing is a checkbox to select all at the same time. I uploaded a bunch of Manga scans, and wanted to edit the metadata of 30 of them at the same time. Filtering shows only the 30, but then I need to select one by one. 

Also, would be great to have the possibility of fetching 1 metadata, lets say for volume 1 and then replicating it to all selected in such situations.

**Reply by cosmintataru** [reply]
I was looking for a checkbox or button to select all books listed on the page as well, **please**. I'd need to delete a large number of books in bulk. I'd be very happy to see this implemented.
Thank you in advance

**Comment by si-remo** [comment]
An "Add selected books to shelf" option would be absolutely fantastic.
Alternatively. Add "Shelf" option in the "Edit Selected Books" popup frame.
